src/app/authoring-tool/select-component/select-component.component.ts

```diff
@@ -0,0 +1,66 @@
+import { CommonModule } from '@angular/common';
+import { Component, EventEmitter, Input, Output, SimpleChanges } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatSelectModule } from '@angular/material/select';
+import { ProjectService } from '../../../assets/wise5/services/projectService';
+import { ComponentContent } from '../../../assets/wise5/common/ComponentContent';
+
+@Component({
+  selector: 'select-component',
+  templateUrl: './select-component.component.html',
+  standalone: true,
+  imports: [CommonModule, FormsModule, MatFormFieldModule, MatSelectModule]
+})
+export class SelectComponentComponent {
+  @Input() allowedComponentTypes: string[] = [];
+  @Output() componentChangedEvent: EventEmitter<string> = new EventEmitter<string>();
+  @Input() componentId: string;
+  protected components: ComponentContent[] = [];
+  protected componentToIsAllowed: Map<string, boolean> = new Map<string, boolean>();
+  @Input() nodeId: string;
+  @Input() thisComponentId: string;
+
+  constructor(private projectService: ProjectService) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.nodeId) {
+      this.nodeId = changes.nodeId.currentValue;
+      this.componentId = null;
```

Comment:
Instead of setting componentId to null here, how about:
1. Rename automaticallySetComponentIfPossible() to setComponentId()
2. In setComponent(), set componentId to null if you can't automatically set it.